java/meterpreter/meterpreter/src/main/java/com/metasploit/meterpreter/Transport.java

```diff
@@ -161,8 +162,8 @@ protected void encodePacketAndWrite(TLVPacket tlvPacket, int type, DataOutputStr
         if (this.aesKey != null) {
             try
             {
-                if (this.aesEnabled) {
-                    encType = ENC_AES256;
+                if (this.aesEnabled && this.aesKey != null) {
```

Comment:
Very minor but we don't need the null check here, it's already checked for null on line 161. Let me fix this up on landing